File: BinaryTreeLevelOrderTraversalII/third.java
The following is my AC solution Binary Tree Level Order Traversal II (II). It is only one difference from Binary Tree Level Order Traversal I (I). Note that for II, we do result.add(0, level), so every time the level is added at the beginning (index 0). For I, we only change this line to result.add(level). Simple!

public class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        List result = new ArrayList();
        if (root == null) {
            return result;
        }
        Queue<TreeNode> queue = new LinkedList<TreeNode>();
        queue.offer(root);

        while (!queue.isEmpty()) {
            ArrayList<Integer> level = new ArrayList();
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                TreeNode head = queue.poll();
                level.add(head.val);
                if (head.left != null) {
                  queue.offer(head.left);
                }
                if (head.right != null) {
                    queue.offer(head.right);
                }
            }
            result.add(0, level);
        }

        return result;
    }
}

File: BinaryTreeLevelOrderTraversalII/third.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> nodes;
        if(!root)
            return nodes;

        stack<vector<int>> s;
        queue<TreeNode*> q;
        q.push(root);

        while(1){
            int nodecount = q.size();
            vector<int> levelnodes;
            if(nodecount == 0)
                break;

            while(nodecount > 0){
                TreeNode *p = q.front();
                q.pop();
                levelnodes.push_back(p->val);
                nodecount--;
                if(p->left)
                    q.push(p->left);
                if(p->right)
                    q.push(p->right);
            }
            // one level complete
            s.push(levelnodes);
        }
        // now push elements to the nodes vector
        while(s.size()){
            nodes.push_back(s.top());
            s.pop();
        }


        return nodes;

    }    
};`
